### PR TITLE
Feature/39711 gias bulk import

### PR DESCRIPTION
**Pull request by jon-shipley**
School import 

* maps required columns dynamically
* input file must be manually uploaded to Storage: school-import and end in '.csv'
* stdout & stderr will be uploaded to blob storage (stderr is optional)
* estimated run time is 20 seconds

ToDo

 - [x] Deal with the spec change

### CODE REVIEW

File: admin/data/sql/migrations/20200225171914.do.bulk-import-user-1.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+'use strict'
+
+const config = require('../../../config')
+
+module.exports.generateSql = function () {
+  return `   
+    -- CREATE SCHEMA [mtc_census_import];
```
Comment (GeorgeChatzigiannis) [inline admin/data/sql/migrations/20200225171914.do.bulk-import-user-1.js]: Any particular reason for keeping these commented lines?

File: admin/data/sql/migrations/20200225171914.undo.bulk-import-user-1.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+'use strict'
+
+const config = require('../../../config')
+
+module.exports.generateSql = function () {
+  return `   
+    -- CREATE SCHEMA [mtc_census_import];
```
Comment (GeorgeChatzigiannis) [inline admin/data/sql/migrations/20200225171914.undo.bulk-import-user-1.js]: as above

File: functions-app/school-import/service/data-access/predicates.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
+'use strict'
+const RA = require('ramda-adjunct')
+
+const predicates = {
+  isSchoolOpen (logger, school) {
+    // we want to load all schools that are open, proposed to open, proposed to close
+    // this is the same as every school that isn't closed
+    if (school.estabStatusCode === '2') {
```
Comment (GeorgeChatzigiannis) [inline functions-app/school-import/service/data-access/predicates.js]: magic number? perhaps introduce a const to make this if statement a bit more readable?

File: admin/data/sql/migrations/20200225171914.do.bulk-import-user-1.js (hunk 1 above)
Comment (GeorgeChatzigiannis) [reply admin/data/sql/migrations/20200225171914.do.bulk-import-user-1.js]: ignore this it has been updated

File: functions-app/school-import/service/data-access/school.data.service.js [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+    table.columns.add('urn', mssql.Int, { nullable: false })
+
+    for (let i = 0; i < data.length; i++) {
+      const mapped = this.getMappedData(data[i], mapping)
+      meta.linesProcessed += 1
+
+      if (this.isPredicated(mapped)) {
+        meta.schoolsLoaded += 1
+        const dfeNumber = parseInt('' + mapped.leaCode + mapped.estabCode, 10)
+        if (dfeNumber.toString().length !== 7) {
```
Comment (GuyHarwood) [inline functions-app/school-import/service/data-access/school.data.service.js]: The message suggests an exception would always be short but could it be longer?

File: functions-app/school-import/service/data-access/predicates.js (hunk 3 above)
Comment (jon-shipley) [reply functions-app/school-import/service/data-access/predicates.js]: Good spot.  Updated.

File: functions-app/school-import/service/data-access/school.data.service.js (hunk 4 above)
Comment (jon-shipley) [reply functions-app/school-import/service/data-access/school.data.service.js]: fixed

File: functions-app/school-import/index.js [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    meta = await v1.process(context, blob)
+  } catch (error) {
+    meta = { processCount: 'n/a' }
+    context.log.error(`${name}: ERROR: ${error.message}`, error)
+  }
+
+  const end = performance.now()
+  const durationInMilliseconds = end - start
+  const timeStamp = new Date().toISOString()
+  context.log(`${name}: ${timeStamp} processed ${meta.processCount} checks, run took ${durationInMilliseconds} ms`)
```
Comment (GuyHarwood) [inline functions-app/school-import/index.js]: schools instead of checks, as we noticed

File: functions-app/school-import/readme.md [hunk 6]
```diff
@@ -0,0 +1,14 @@
+# BlobTrigger - JavaScript
+
+The `BlobTrigger` makes it incredibly easy to react to new Blobs inside of Azure Blob Storage. This sample demonstrates
```
Comment (GuyHarwood) [inline functions-app/school-import/readme.md]: MS Marketing blurb alert 🔔

Comment (jon-shipley) [reply functions-app/school-import/readme.md]: @GuyHarwood Good spot!  I've updated the readme with something more useful. ⚔️

**Pull request merged**